Sources/XcodeProj/Objects/Project/PBXProj.swift

```diff
@@ -52,6 +52,24 @@ public final class PBXProj: Decodable {
         }
     }
 
+    /// Initializes the project with a path to the pbxproj file.
+    ///
+    /// - Parameters:
+    ///   - path: Path to a pbxproj file.
+    public convenience init(path: Path) throws {
+        let pbxproject: PBXProj = try PBXProj.createPBXProj(path: path)
+        var objects: [PBXObject] = []
+        pbxproject.objects.forEach { object in
+            objects.append(object)
+        }
```

Comment:
I wonder if this has any performance impact on large projects 🤔

Looking at the code it seems the array `[PBXObject]` is only transiently needed to pass to the public initialiser, but later gets re-assembled as `PBXObjects` _(which internally loops over the objects to process them)_ - this seems like an unnecessary roundtrip 😅

We may be able to resolve this by:

- making this a regular initialiser and directly assign the variables from loaded `pbxproj`
- creating another private initialiser that can be provided with `PBXObjects` directly to avoid needing an extra hop of making the `[PBXObject]`